test/e2e/framework/manifest/manifest.go

```diff
@@ -85,3 +91,48 @@ func StatefulSetFromManifest(fileName, ns string) (*appsv1.StatefulSet, error) {
 	}
 	return &ss, nil
 }
+
+// GetDaemonSetFromURL reads from a url and returns the daemonset in it.
+func GetDaemonSetFromURL(url string) (*appsv1.DaemonSet, error) {
+	framework.Logf("Parsing ds from %v", url)
+
+	var response *http.Response
+	var err error
+
+	for i := 1; i <= 5; i++ {
+		response, err = http.Get(url)
+		if err == nil && response.StatusCode == 200 {
+			break
+		}
+		time.Sleep(time.Duration(i) * time.Second)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get url: %v", err)
+	}
+	if response.StatusCode != 200 {
+		return nil, fmt.Errorf("invalid http response status: %v", response.StatusCode)
+	}
+	defer response.Body.Close()
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read html response body: %v", err)
+	}
+	return DsFromData(data)
+}
+
+// DsFromData reads a byte slice and returns the daemonset in it.
+func DsFromData(data []byte) (*appsv1.DaemonSet, error) {
```

Comment:
If renaming the above function to `DaemonSetFromURL`, this function could be renamed to `DaemonSetFromData`.